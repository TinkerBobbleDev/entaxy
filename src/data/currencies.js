export const fiatCurrencies = {
  EUR: 'Euro',
  USD: 'US dollar',
  JPY: 'Japanese Yen',
  BGN: 'Bulgarian Lev',
  CZK: 'Czech Koruna',
  DKK: 'Danish Krone',
  GBP: 'British Pound',
  HUF: 'Hungarian Forint',
  PLN: 'Polish Zloty',
  RON: 'Romanian Leu',
  SEK: 'Swedish Krona',
  CHF: 'Swiss Franc',
  ISK: 'Icelandic Kr\u00f3na',
  NOK: 'Norwegian Krone',
  HRK: 'Croatian Kuna',
  RUB: 'Russian Ruble',
  TRY: 'Turkish Lira',
  AUD: 'Australian Dollar',
  BRL: 'Brazilian Real',
  CAD: 'Canadian Dollar',
  CNY: 'Chinese Yuan',
  HKD: 'Hong Kong Dollar',
  IDR: 'Indonesian Rupiah',
  ILS: 'Israeli New Shekel',
  INR: 'Indian Rupee',
  KRW: 'South Korean Won',
  MXN: 'Mexican Peso',
  MYR: 'Malaysian Ringgit',
  NZD: 'New Zealand Dollar',
  PHP: 'Philippine Peso',
  SGD: 'Singapore Dollar',
  THB: 'Thai Baht',
  ZAR: 'South African Rand'
}

export const cryptoCurrencies = {
  BTN: 'Bitcoin',
  BCH: 'Bitcoin Cash',
  LTC: 'Litecoin',
  ETH: 'Ethereum'
}

export const formatedFiatCurrencies = Object.keys(fiatCurrencies)
  .sort()
  .map(key => ({
    value: key,
    label: `(${key}) ${fiatCurrencies[key]}`
  }))

export const filteredFiatCurrencies = (inputValue) => {
  return new Promise((resolve) => {
    if (inputValue) {
      resolve(formatedFiatCurrencies.filter(
        currency => currency.label.toLowerCase().includes(inputValue.toLowerCase())
      ))
    }
    resolve(formatedFiatCurrencies)
  })
}
