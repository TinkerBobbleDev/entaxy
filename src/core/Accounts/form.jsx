import React from 'react'
import { compose } from 'recompose'
import { connect } from 'react-redux'
import { withStyles } from '@material-ui/core/styles'
import { withFormik } from 'formik'
import * as Yup from 'yup'
import PropTypes from 'prop-types'
import Grid from '@material-ui/core/Grid'
import Paper from '@material-ui/core/Paper'
import Typography from '@material-ui/core/Typography'
import TextField from '@material-ui/core/TextField'
import IconButton from '@material-ui/core/IconButton'
import CloseIcon from '@material-ui/icons/Close'
import Divider from '@material-ui/core/Divider'
import Button from '@material-ui/core/Button'
import red from '@material-ui/core/colors/red'
import format from 'date-fns/format'
import parse from 'date-fns/parse'
import { fiatCurrencies, filteredFiatCurrencies } from '../../data/currencies'
import AutoComplete from '../../common/AutoComplete'
import institutions from '../../data/institutions'
import SubmitButtonWithProgress from '../../common/SubmitButtonWithProgress'
import DescriptionCard from '../../common/DescriptionCard'
import LinkTo from '../../common/LinkTo'

const styles = theme => ({
  root: {
    margin: theme.spacing(2),
    padding: theme.spacing(2)
  },
  formHeader: {
    padding: 10,
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-between'
  },
  closeButton: {
    marginTop: -10
  },
  form: {
    display: 'flex',
    flexDirection: 'column'
  },
  input: {
    margin: theme.spacing(2),
    width: 320
  },
  inputError: {
    marginTop: -7,
    marginLeft: theme.spacing(2)
  },
  formActions: {
    display: 'flex',
    justifyContent: 'flex-end',
    paddingTop: 10
  },
  deleteButton: {
    color: red[500],
    margin: theme.spacing(2)
  }
})

const mapStateToProps = (state, ownProps) => ({
  settings: state.settings,
  accountInstitutions: Object.keys(state.accounts.byInstitution),
  account: state.accounts.byId[ownProps.accountId],
  accounts: state.accounts
})

export class AccountFormComponent extends React.Component {
  state = {
    hideInstitutionOptions: false
  }

  institutionOptions = (institution) => {
    const { classes } = this.props
    const { hideInstitutionOptions } = this.state
    const { value } = (institution || {})

    if (!Object.keys(institutions).includes(value) || hideInstitutionOptions) return null

    if (institutions[value].importTypes.includes('API')) {
      return (
        <DescriptionCard
          info
          className={classes.input}
          actions={(
            <Grid align="center">
              <Button
                size="small"
                color="secondary"
                component={LinkTo(`/institutions/${value}/import/new`)}
              >
                Great, Let&apos;s do it
              </Button>
              <Button
                size="small"
                color="primary"
                onClick={this.handleCloseInstitutionOptions}
              >
                No, I&apos;d rather do it manually
              </Button>
            </Grid>
          )}
        >
          <Typography variant="caption" paragraph>
            You can import&nbsp;
            <strong>all your accounts</strong>
            &nbsp;in one go from&nbsp;
            <strong>{institutions[value].name}</strong>
            &nbsp;by using their API.
          </Typography>
          <Typography variant="caption" align="center">
            This is also the easiest way to keep your transactions up to date.
          </Typography>
        </DescriptionCard>
      )
    }
    return null
  }

  handleInstitutionChange = (...args) => {
    this.setState({ hideInstitutionOptions: false })
    this.props.setFieldValue(...args)
  }

  handleCloseInstitutionOptions = () => {
    this.setState({ hideInstitutionOptions: true })
  }


  formatedInstitutions = () => {
    const allInstitutions = new Set(this.props.accountInstitutions.concat(Object.keys(institutions)))
    return Array.from(allInstitutions).sort().map(key => ({ value: key, label: key }))
  }

  render() {
    const {
      classes,
      handleSubmit,
      isSubmitting,
      values,
      errors,
      touched,
      handleChange,
      setFieldValue,
      handleDelete,
      handleCancel,
      account
    } = this.props
    return (
      <Grid container direction="row" justify="center">
        <Paper className={classes.root}>
          <div className={classes.formHeader}>
            <Typography variant="h6">
              {account ? 'Edit account' : 'New account'}
            </Typography>
            <IconButton aria-label="Close" className={classes.closeButton} onClick={handleCancel}>
              <CloseIcon />
            </IconButton>
          </div>
          <Divider />
          <form onSubmit={handleSubmit} className={classes.form}>
            <AutoComplete
              creatable
              className={classes.input}
              label="Institution"
              name="institution"
              value={values.institution}
              options={this.formatedInstitutions()}
              onChange={this.handleInstitutionChange}
              error={errors.institution && touched.institution}
              helperText={errors.institution}
              autoFocus
            />
            {this.institutionOptions(values.institution)}
            <TextField
              className={classes.input}
              label="Account name"
              inputProps={{
                'aria-label': 'Account name',
                maxLength: 100
              }}
              value={values.name}
              name="name"
              onChange={handleChange}
              error={errors.name && touched.name}
              helperText={errors.name}
            />
            <TextField
              type="number"
              className={classes.input}
              label="Opening balance"
              inputProps={{
                'aria-label': 'Opening balance',
                step: 0.01
              }}
              value={values.openingBalance}
              name="openingBalance"
              onChange={handleChange}
              error={errors.openingBalance && touched.openingBalance}
              helperText={errors.openingBalance}
            />
            <TextField
              type="date"
              label="Opening Balance Date"
              InputLabelProps={{
                shrink: true,
                'aria-label': 'Opening Balance Date'
              }}
              name="openingBalanceDate"
              className={classes.input}
              value={values.openingBalanceDate}
              onChange={handleChange}
              error={errors.openingBalanceDate && touched.openingBalanceDate}
              helperText={errors.openingBalanceDate}
            />
            <AutoComplete
              async
              className={classes.input}
              label="Account Currency"
              name="currency"
              value={values.currency}
              loadOptions={filteredFiatCurrencies}
              onChange={setFieldValue}
              error={errors.currency && touched.currency}
              helperText={errors.currency}
            />
            {handleDelete && (
              <Button
                size="small"
                onClick={() => handleDelete(account)}
                className={classes.deleteButton}
                disabled={isSubmitting}
              >
                Delete this account
              </Button>
            )}
            <Divider />
            <div className={classes.formActions}>
              <SubmitButtonWithProgress label="Save" isSubmitting={isSubmitting} />
            </div>
          </form>
        </Paper>
      </Grid>
    )
  }
}

AccountFormComponent.propTypes = {
  classes: PropTypes.object.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  isSubmitting: PropTypes.bool.isRequired,
  values: PropTypes.object.isRequired,
  errors: PropTypes.object.isRequired,
  touched: PropTypes.object.isRequired,
  handleChange: PropTypes.func.isRequired,
  setFieldValue: PropTypes.func.isRequired,
  handleDelete: PropTypes.func,
  handleCancel: PropTypes.func.isRequired,
  accountInstitutions: PropTypes.array.isRequired,
  account: PropTypes.object
}

AccountFormComponent.defaultProps = {
  handleDelete: null,
  account: null
}

export default compose(
  connect(mapStateToProps),
  withStyles(styles),
  withFormik({
    enableReinitialize: true,
    mapPropsToValues: ({ account, settings }) => {
      if (account === undefined) {
        return {
          name: '',
          institution: null,
          openingBalance: 0,
          openingBalanceDate: format(Date.now(), 'YYYY-MM-DD'),
          currency: settings.currency === undefined ? null : {
            label: `(${settings.currency}) ${fiatCurrencies[settings.currency]}`,
            value: settings.currency
          }
        }
      }
      return {
        ...account,
        openingBalanceDate: format(new Date(account.openingBalanceDate), 'YYYY-MM-DD'),
        institution: {
          label: account.institution,
          value: account.institution
        },
        currency: {
          label: `(${account.currency}) ${fiatCurrencies[account.currency]}`,
          value: account.currency
        }
      }
    },
    validationSchema: (props) => {
      return Yup.lazy((values) => {
        const accountNames = Object.values(props.accounts.byId).filter(
          account => account.institution === values.institution.value
        ).map(account => account.name)
        console.log(accountNames, values)
        return Yup.object().shape({
          name: Yup.string()
            .max(50, 'Too Long!')
            .required('Please enter a name for this account')
            .notOneOf(accountNames, `There's already an account with this name in ${values.institution.value}`),
          institution: Yup.object()
            .required('Please select an institution')
            .nullable(accountNames),
          openingBalance: Yup.number()
            .required('Please enter an opening balance')
            .min(-999999999.99)
            .max(999999999.99),
          openingBalanceDate: Yup.date()
            .required('Please select the date of the opening balance'),
          currency: Yup.object()
            .required('Please select the currency of this account')
            .nullable()
        })
      })
    },
    handleSubmit: async (values, { props, setSubmitting }) => {
      setSubmitting(true)
      await props.handleSave({
        ...values,
        institution: values.institution.value,
        openingBalanceDate: parse(values.openingBalanceDate).getTime(),
        currency: values.currency.value
      })
      setSubmitting(false)
    }
  })
)(AccountFormComponent)
