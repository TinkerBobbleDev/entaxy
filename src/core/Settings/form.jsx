import React from 'react'
import { compose } from 'recompose'
import { connect } from 'react-redux'
import { withStyles } from '@material-ui/core/styles'
import { withFormik } from 'formik'
import PropTypes from 'prop-types'
import Typography from '@material-ui/core/Typography'
import Divider from '@material-ui/core/Divider'
import Button from '@material-ui/core/Button'
import CircularProgress from '@material-ui/core/CircularProgress'
import red from '@material-ui/core/colors/red'
import green from '@material-ui/core/colors/green'
import AsyncSelect from 'react-select/lib/Async'
import currencies from '../../data/currencies'
import locales from '../../data/locales'
import confirm from '../../util/confirm'

const styles = theme => ({
  form: {
    display: 'flex',
    flexDirection: 'column'
  },
  inputTitle: {
    marginLeft: theme.spacing.unit * 2,
    marginTop: theme.spacing.unit * 2,
    marginBottom: -theme.spacing.unit * 2
  },
  input: {
    margin: theme.spacing.unit * 2,
    width: 320
  },
  formActions: {
    display: 'flex',
    justifyContent: 'flex-end',
    paddingTop: 10
  },
  deleteButton: {
    color: red[500],
    margin: theme.spacing.unit * 2
  },
  buttonWrapper: {
    margin: theme.spacing.unit,
    position: 'relative'
  },
  buttonProgress: {
    color: green[500],
    position: 'absolute',
    top: '50%',
    left: '50%',
    marginTop: -12,
    marginLeft: -12
  },
  close: {
    padding: theme.spacing.unit / 2
  }
})

const mapStateToProps = ({ settings }) => ({ settings })

const formatedLocales = Object.keys(locales).map(key => ({ value: key, label: locales[key] }))
const formatedCurrencies = Object.keys(currencies).map(key => ({ value: key, label: `(${key}) ${currencies[key]}` }))

export class SettingsFormComponent extends React.Component {
  filteredCurrencies = (inputValue) => {
    return new Promise((resolve) => {
      if (inputValue) {
        resolve(formatedCurrencies.filter(currency => currency.label.toLowerCase().includes(inputValue.toLowerCase())))
      }
      resolve(formatedCurrencies)
    })
  }

  filteredLocales = (inputValue) => {
    return new Promise((resolve) => {
      if (inputValue) {
        resolve(formatedLocales.filter(locale => locale.label.toLowerCase().includes(inputValue.toLowerCase())))
      }
      resolve(formatedLocales)
    })
  }

  handleResetData = () => {
    confirm('Delete all your data? This cannot be undone.', 'Are you sure?').then(() => {
      this.props.handleDeleteAllData()
    })
  }

  render() {
    const {
      classes,
      handleSubmit,
      values,
      setFieldValue,
      isSubmitting
    } = this.props

    return (
      <form onSubmit={handleSubmit} className={classes.form}>
        <Typography variant="subtitle2" className={classes.inputTitle}>Language / Country</Typography>
        <AsyncSelect
          label="Language / Country"
          name="locale"
          value={values.locale}
          cacheOptions
          defaultOptions
          loadOptions={this.filteredLocales}
          onChange={selection => setFieldValue('locale', selection)}
          className={classes.input}
          isClearable={true}
          inputProps={{ 'aria-label': 'Language / Country', required: true }}
        />
        <Typography variant="subtitle2" className={classes.inputTitle}>Display Currency</Typography>
        <AsyncSelect
          label="Display Currency"
          name="currency"
          value={values.currency}
          cacheOptions
          loadOptions={this.filteredCurrencies}
          onChange={selection => setFieldValue('currency', selection)}
          className={classes.input}
          isClearable={true}
          inputProps={{ 'aria-label': 'Display Currency', required: true }}
        />
        <Button
          size="small"
          onClick={this.handleResetData}
          className={classes.deleteButton}
        >
          Reset (Delete all my data)
        </Button>
        <Divider />
        <div className={classes.formActions}>
          <div className={classes.buttonWrapper}>
            <Button type="submit" color="secondary" disabled={isSubmitting}>Save</Button>
            {isSubmitting && <CircularProgress size={24} className={classes.buttonProgress} />}
          </div>
        </div>
      </form>
    )
  }
}

SettingsFormComponent.propTypes = {
  classes: PropTypes.object.isRequired,
  handleDeleteAllData: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  values: PropTypes.object.isRequired,
  setFieldValue: PropTypes.func.isRequired,
  isSubmitting: PropTypes.bool.isRequired
}

export default compose(
  connect(mapStateToProps),
  withStyles(styles),
  withFormik({
    mapPropsToValues: ({ settings }) => {
      return ({
        ...settings,
        locale: {
          label: locales[settings.locale],
          value: settings.locale
        },
        currency: settings.currency === undefined ? '' : {
          label: currencies[settings.currency],
          value: settings.currency
        }
      })
    },
    handleSubmit: (values, { props, setSubmitting }) => {
      setSubmitting(true)
      props.handleSave({
        ...values,
        locale: values.locale.value,
        currency: values.currency.value
      }).then(() => {
        setSubmitting(false)
      })
    }
  })
)(SettingsFormComponent)
