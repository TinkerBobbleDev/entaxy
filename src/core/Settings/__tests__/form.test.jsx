import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import { SettingsFormComponent } from '../form'
import currencies from '../../../data/currencies'
import locales from '../../../data/locales'

const mochHandleSave = jest.fn()
const mochHandleDeleteAllData = jest.fn()
const mochHandleSubmit = jest.fn()
const mochHandleChange = jest.fn()
const mochSetFieldValue = jest.fn()

describe('SettingsForm', () => {
  it('matches snapshot before submitting form', () => {
    const component = renderer.create((
      <SettingsFormComponent
        handleSave={mochHandleSave}
        handleDeleteAllData={mochHandleDeleteAllData}
        classes={{}}
        handleSubmit={mochHandleSubmit}
        values={{}}
        handleChange={mochHandleChange}
        setFieldValue={mochSetFieldValue}
        isSubmitting={false}
      />
    ))
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('matches snapshot before after submitting form', () => {
    const component = renderer.create((
      <SettingsFormComponent
        handleSave={mochHandleSave}
        handleDeleteAllData={mochHandleDeleteAllData}
        classes={{}}
        handleSubmit={mochHandleSubmit}
        values={{}}
        handleChange={mochHandleChange}
        setFieldValue={mochSetFieldValue}
        isSubmitting={true}
      />
    ))
    expect(component.toJSON()).toMatchSnapshot()
  })

  describe('Component methods', () => {
    const wrapper = shallow((
      <SettingsFormComponent
        handleSave={mochHandleSave}
        handleDeleteAllData={mochHandleDeleteAllData}
        classes={{}}
        handleSubmit={mochHandleSubmit}
        values={{}}
        handleChange={mochHandleChange}
        setFieldValue={mochSetFieldValue}
        isSubmitting={true}
      />
    ))
    const instance = wrapper.instance()

    describe('filteredCurrencies', () => {
      it('filters with no input', async () => {
        const result = await instance.filteredCurrencies()
        expect(result).toEqual(Object.keys(currencies).map(key => ({
          value: key,
          label: `(${key}) ${currencies[key]}`
        })))
      })

      it('filters with some input', async () => {
        const result = await instance.filteredCurrencies('EU')
        expect(result).toEqual([
          { label: '(EUR) Euro', value: 'EUR' },
          { label: '(XEU) European Currency Unit', value: 'XEU' },
          { label: '(MDL) Moldovan Leu', value: 'MDL' },
          { label: '(RON) Romanian Leu', value: 'RON' },
          { label: '(ROL) Romanian Leu (1952â€“2006)', value: 'ROL' },
          { label: '(CHE) WIR Euro', value: 'CHE' }
        ])
      })

      it('filters with a specific input', async () => {
        const result = await instance.filteredCurrencies('CAD')
        expect(result).toEqual([{ label: `(CAD) ${currencies.CAD}`, value: 'CAD' }])
      })
    })

    describe('filteredLocales', () => {
      it('filters with no input', async () => {
        const result = await instance.filteredLocales()
        expect(result).toEqual(Object.keys(locales).map(key => ({
          value: key,
          label: locales[key]
        })))
      })

      it('filters with some input', async () => {
        const result = await instance.filteredLocales('Cana')
        expect(result).toEqual([
          { label: 'English (Canada)', value: 'en-CA' },
          { label: 'French (Canada)', value: 'fr-CA' },
          { label: 'Spanish (Canary Islands)', value: 'es-IC' }
        ])
      })

      it('filters with a specific input', async () => {
        const result = await instance.filteredLocales('English (Canada)')
        expect(result).toEqual([{ label: 'English (Canada)', value: 'en-CA' }])
      })
    })
  })
})
