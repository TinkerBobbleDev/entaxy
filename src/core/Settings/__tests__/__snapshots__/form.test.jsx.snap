// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsForm matches snapshot before after submitting form 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <h6
    className="MuiTypography-root-1 MuiTypography-subtitle2-20"
  >
    Language / Country
  </h6>
  <div
    className="css-10nd86i"
    onKeyDown={[Function]}
  >
    <div
      className="css-vj8t7z"
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
    >
      <div
        className="css-1hwfws3"
      >
        <div
          className="css-1492t68"
        >
          Select...
        </div>
        <div
          className="css-1g6gooi"
        >
          <div
            className=""
            style={
              Object {
                "display": "inline-block",
              }
            }
          >
            <input
              aria-autocomplete="list"
              autoCapitalize="none"
              autoComplete="off"
              autoCorrect="off"
              disabled={false}
              id="react-select-4-input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              spellCheck="false"
              style={
                Object {
                  "background": 0,
                  "border": 0,
                  "boxSizing": "content-box",
                  "color": "inherit",
                  "fontSize": "inherit",
                  "opacity": 1,
                  "outline": 0,
                  "padding": 0,
                  "width": "1px",
                }
              }
              tabIndex="0"
              type="text"
              value=""
            />
            <div
              style={
                Object {
                  "height": 0,
                  "left": 0,
                  "overflow": "scroll",
                  "position": "absolute",
                  "top": 0,
                  "visibility": "hidden",
                  "whiteSpace": "pre",
                }
              }
            >
              
            </div>
          </div>
        </div>
      </div>
      <div
        className="css-1wy0on6"
      >
        <div
          aria-hidden="true"
          className="css-fraxkc"
        >
          <span
            className="css-pujnn1"
          />
          <span
            className="css-gji98i"
          />
          <span
            className="css-1frpy8x"
          />
        </div>
        <span
          className="css-d8oujb"
        />
        <div
          aria-hidden="true"
          className="css-1ep9fjw"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <svg
            aria-hidden="true"
            className="css-19bqh2r"
            focusable="false"
            height={20}
            viewBox="0 0 20 20"
            width={20}
          >
            <path
              d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
            />
          </svg>
        </div>
      </div>
    </div>
    <input
      name="locale"
      type="hidden"
      value=""
    />
  </div>
  <h6
    className="MuiTypography-root-1 MuiTypography-subtitle2-20"
  >
    Display Currency
  </h6>
  <div
    className="css-10nd86i"
    onKeyDown={[Function]}
  >
    <div
      className="css-vj8t7z"
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
    >
      <div
        className="css-1hwfws3"
      >
        <div
          className="css-1492t68"
        >
          Select...
        </div>
        <div
          className="css-1g6gooi"
        >
          <div
            className=""
            style={
              Object {
                "display": "inline-block",
              }
            }
          >
            <input
              aria-autocomplete="list"
              autoCapitalize="none"
              autoComplete="off"
              autoCorrect="off"
              disabled={false}
              id="react-select-5-input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              spellCheck="false"
              style={
                Object {
                  "background": 0,
                  "border": 0,
                  "boxSizing": "content-box",
                  "color": "inherit",
                  "fontSize": "inherit",
                  "opacity": 1,
                  "outline": 0,
                  "padding": 0,
                  "width": "1px",
                }
              }
              tabIndex="0"
              type="text"
              value=""
            />
            <div
              style={
                Object {
                  "height": 0,
                  "left": 0,
                  "overflow": "scroll",
                  "position": "absolute",
                  "top": 0,
                  "visibility": "hidden",
                  "whiteSpace": "pre",
                }
              }
            >
              
            </div>
          </div>
        </div>
      </div>
      <div
        className="css-1wy0on6"
      >
        <span
          className="css-d8oujb"
        />
        <div
          aria-hidden="true"
          className="css-1ep9fjw"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <svg
            aria-hidden="true"
            className="css-19bqh2r"
            focusable="false"
            height={20}
            viewBox="0 0 20 20"
            width={20}
          >
            <path
              d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
            />
          </svg>
        </div>
      </div>
    </div>
    <input
      name="currency"
      type="hidden"
      value=""
    />
  </div>
  <button
    className="MuiButtonBase-root-62 MuiButton-root-36 MuiButton-text-38 MuiButton-flat-41 MuiButton-sizeSmall-59"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiButton-label-37"
    >
      Reset (Delete all my data)
    </span>
    <span
      className="MuiTouchRipple-root-70"
    />
  </button>
  <hr
    className="MuiDivider-root-65"
  />
  <div>
    <div>
      <button
        className="MuiButtonBase-root-62 MuiButtonBase-disabled-63 MuiButton-root-36 MuiButton-text-38 MuiButton-textSecondary-40 MuiButton-flat-41 MuiButton-flatSecondary-43 MuiButton-disabled-56"
        disabled={true}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="-1"
        type="submit"
      >
        <span
          className="MuiButton-label-37"
        >
          Save
        </span>
      </button>
      <div
        className="MuiCircularProgress-root-77 MuiCircularProgress-colorPrimary-80 MuiCircularProgress-indeterminate-79"
        role="progressbar"
        style={
          Object {
            "height": 24,
            "width": 24,
          }
        }
      >
        <svg
          className="MuiCircularProgress-svg-82"
          viewBox="22 22 44 44"
        >
          <circle
            className="MuiCircularProgress-circle-83 MuiCircularProgress-circleIndeterminate-85"
            cx={44}
            cy={44}
            fill="none"
            r={20.2}
            strokeWidth={3.6}
            style={Object {}}
          />
        </svg>
      </div>
    </div>
  </div>
</form>
`;

exports[`SettingsForm matches snapshot before submitting form 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <h6
    className="MuiTypography-root-1 MuiTypography-subtitle2-20"
  >
    Language / Country
  </h6>
  <div
    className="css-10nd86i"
    onKeyDown={[Function]}
  >
    <div
      className="css-vj8t7z"
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
    >
      <div
        className="css-1hwfws3"
      >
        <div
          className="css-1492t68"
        >
          Select...
        </div>
        <div
          className="css-1g6gooi"
        >
          <div
            className=""
            style={
              Object {
                "display": "inline-block",
              }
            }
          >
            <input
              aria-autocomplete="list"
              autoCapitalize="none"
              autoComplete="off"
              autoCorrect="off"
              disabled={false}
              id="react-select-2-input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              spellCheck="false"
              style={
                Object {
                  "background": 0,
                  "border": 0,
                  "boxSizing": "content-box",
                  "color": "inherit",
                  "fontSize": "inherit",
                  "opacity": 1,
                  "outline": 0,
                  "padding": 0,
                  "width": "1px",
                }
              }
              tabIndex="0"
              type="text"
              value=""
            />
            <div
              style={
                Object {
                  "height": 0,
                  "left": 0,
                  "overflow": "scroll",
                  "position": "absolute",
                  "top": 0,
                  "visibility": "hidden",
                  "whiteSpace": "pre",
                }
              }
            >
              
            </div>
          </div>
        </div>
      </div>
      <div
        className="css-1wy0on6"
      >
        <div
          aria-hidden="true"
          className="css-fraxkc"
        >
          <span
            className="css-pujnn1"
          />
          <span
            className="css-gji98i"
          />
          <span
            className="css-1frpy8x"
          />
        </div>
        <span
          className="css-d8oujb"
        />
        <div
          aria-hidden="true"
          className="css-1ep9fjw"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <svg
            aria-hidden="true"
            className="css-19bqh2r"
            focusable="false"
            height={20}
            viewBox="0 0 20 20"
            width={20}
          >
            <path
              d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
            />
          </svg>
        </div>
      </div>
    </div>
    <input
      name="locale"
      type="hidden"
      value=""
    />
  </div>
  <h6
    className="MuiTypography-root-1 MuiTypography-subtitle2-20"
  >
    Display Currency
  </h6>
  <div
    className="css-10nd86i"
    onKeyDown={[Function]}
  >
    <div
      className="css-vj8t7z"
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
    >
      <div
        className="css-1hwfws3"
      >
        <div
          className="css-1492t68"
        >
          Select...
        </div>
        <div
          className="css-1g6gooi"
        >
          <div
            className=""
            style={
              Object {
                "display": "inline-block",
              }
            }
          >
            <input
              aria-autocomplete="list"
              autoCapitalize="none"
              autoComplete="off"
              autoCorrect="off"
              disabled={false}
              id="react-select-3-input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              spellCheck="false"
              style={
                Object {
                  "background": 0,
                  "border": 0,
                  "boxSizing": "content-box",
                  "color": "inherit",
                  "fontSize": "inherit",
                  "opacity": 1,
                  "outline": 0,
                  "padding": 0,
                  "width": "1px",
                }
              }
              tabIndex="0"
              type="text"
              value=""
            />
            <div
              style={
                Object {
                  "height": 0,
                  "left": 0,
                  "overflow": "scroll",
                  "position": "absolute",
                  "top": 0,
                  "visibility": "hidden",
                  "whiteSpace": "pre",
                }
              }
            >
              
            </div>
          </div>
        </div>
      </div>
      <div
        className="css-1wy0on6"
      >
        <span
          className="css-d8oujb"
        />
        <div
          aria-hidden="true"
          className="css-1ep9fjw"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <svg
            aria-hidden="true"
            className="css-19bqh2r"
            focusable="false"
            height={20}
            viewBox="0 0 20 20"
            width={20}
          >
            <path
              d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
            />
          </svg>
        </div>
      </div>
    </div>
    <input
      name="currency"
      type="hidden"
      value=""
    />
  </div>
  <button
    className="MuiButtonBase-root-62 MuiButton-root-36 MuiButton-text-38 MuiButton-flat-41 MuiButton-sizeSmall-59"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiButton-label-37"
    >
      Reset (Delete all my data)
    </span>
    <span
      className="MuiTouchRipple-root-70"
    />
  </button>
  <hr
    className="MuiDivider-root-65"
  />
  <div>
    <div>
      <button
        className="MuiButtonBase-root-62 MuiButton-root-36 MuiButton-text-38 MuiButton-textSecondary-40 MuiButton-flat-41 MuiButton-flatSecondary-43"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="submit"
      >
        <span
          className="MuiButton-label-37"
        >
          Save
        </span>
        <span
          className="MuiTouchRipple-root-70"
        />
      </button>
    </div>
  </div>
</form>
`;
